### ISSUE TITLE
How to annotate panorama

### ISSUE THREAD

**Issue by dengzhijie** [sunset1995/pytorch-layoutnet#8]
Hello
First, thanks for your sharing .
We want to use new indoor panorama samples for training, but how to annotate panorama as shown in the appendix becomes a trouble for us. Could you share the method or some tools to do the work?
Thank you!
![camera_0bc03a051e9d4bed8e9191e263142f17_office_9_frame_equirectangular_domain_](https://user-images.githubusercontent.com/17869984/47889089-20f7e880-de83-11e8-825b-b141f152d3dc.png)
![camera_0bc03a051e9d4bed8e9191e263142f17_office_9_frame_equirectangular_domain_](https://user-images.githubusercontent.com/17869984/47889096-2d7c4100-de83-11e8-91ab-7a1c55c0569c.png)

**Comment by sunset1995** [comment]
Hi,
You could try https://github.com/SunDaDenny/PanoAnnotator which is developed in python.
You should get the corner position on the image annotated by above tool.
Constructing boundary/corner probability map from corner position is simple in mathematic.

Good luck ;)

**Comment by dengzhijie** [comment]
@sunset1995 
Really thanks  for your reply!
I will try this way later.In addition,I have another question to consult you.When I use the official model to process panoramic pictures，sometimes we get the result of a corner migration  as shown in the annex. How can I get a  better result?Can I get a good result after  3D layout optimization?
Thanks much!
Good luck
![112_bon](https://user-images.githubusercontent.com/17869984/47904276-dfd1f980-debf-11e8-8990-d94a8d802ee7.png)

**Comment by dengzhijie** [comment]
@sunset1995 Thanks! 👍 :)

**Comment by glacierwl** [comment]
I have got the corner id position, but I don't know how to get the boundary/corner probability map like the dataset in the repository.Can you tell me ?Thank you @sunset1995

**Comment by sunset1995** [comment]
Hi @glacierwl,

Maybe you can trace the original repo [LayoutNet](https://github.com/zouchuhang/LayoutNet) (I'm just simply downloaded the processed datas without processing myself). 

Or you can tried my new project on layout estimation:  [HorizonNet](https://github.com/sunset1995/HorizonNet).
For training/finetuning on your own dataset, simply put all image under `data/your_datas/*.png` and the corner positions on the equirectangular image under `data/your_datas/*.txt`.
All images should in the size of `512 x 1024` and below is an example of the corners txt:
```
123 115
123 378
426 141
426 353
680 180
680 318
824 175
824 323
```
where the first column is x positions and second column is y position on your image.

**Comment by glacierwl** [comment]
Actually,I have tested my dataset(panorama+corner position) using[ HorizonNet](https://github.com/sunset1995/HorizonNet) before. When I want to test them using [LayoutNet](https://github.com/zouchuhang/LayoutNet
), I get stuck in the puzzle about the conversion from corners txt to edge map and corner map. I have found some codes in the original repo,Thank you.@sunset1995

**Comment by glacierwl** [comment]
I have generated edge map and corner map from label_cor.txt and panorama.Revise some details in this code.[https://github.com/zouchuhang/LayoutNet/blob/master/matlab/preprocessPano.m](url)

**Comment by glacierwl** [comment]
I have generated edge map and corner map from label_cor.txt and panorama.Revise some details in this code[https://github.com/zouchuhang/LayoutNet/blob/master/matlab/preprocessPano.m](url).

**Comment by glacierwl** [comment]
I have generated edge map and corner map from label.txt and panorama.Just revise some codes in this link.
[https://github.com/zouchuhang/LayoutNet/blob/master/matlab/preprocessPano.m](url)

**Comment by glacierwl** [comment]
I have generated edge map and corner map from label.txt and panorama.Just revise some codes in this[ link](https://github.com/zouchuhang/LayoutNet/blob/master/matlab/preprocessPano.m).
